File: Greedy/452-minimum-number-of-arrows-to-burst-balloons/minimum-number-of-arrows-to-burst-balloons.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort(key=lambda x: x[1])

        count = 1
        high = points[0][1]

        for i in range(1, len(points)):
            if points[i][0] > high:
                high = points[i][1]
                count += 1

        return count

File: Greedy/452-minimum-number-of-arrows-to-burst-balloons/minimum-number-of-arrows-to-burst-balloons.cpp
class Solution {
public:
    int findMinArrowShots(vector<vector<int>>& points) {
        sort(points.begin(), points.end(), [](vector<int>& a, vector<int>& b){
            return a[1] < b[1];
        });

        int count = 1;
        int high = points[0][1];

        for(int i = 1; i < points.size(); i++){
            if(points[i][0] > high){
                high  = points[i][1];
                count++;
            }
        }

        return count;
    }
};